vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py

```diff
@@ -39,16 +41,37 @@ def process_weights_after_loading(self, layer) -> None:
                 logical_widths=layer.logical_widths,
             )
 
+            if is_hip():
+                weight, max_w_scale, input_scale = normalize_e4m3fn_to_e4m3fnuz(
+                    weight=weight,
+                    weight_scale=max_w_scale,
+                    input_scale=layer.input_scale)
+                if input_scale is not None:
+                    layer.input_scale = Parameter(input_scale,
+                                                  requires_grad=False)
+
             layer.weight = Parameter(weight.t(), requires_grad=False)
             layer.weight_scale = Parameter(max_w_scale, requires_grad=False)
 
         # If channelwise, scales are already lined up, so just transpose.
         elif self.strategy == QuantizationStrategy.CHANNEL:
             weight = layer.weight
+
+            if is_hip():
```

Comment:
can we move the is_hip() to a utils file? seem to be similar code in multiple cases

Reply:
We could move the input_scale adjustment into utils, as it is the one common part between the 3 quantization types that support HIP fnuz), but IMO that would make it less explicit, because different parameters of the layer (weight, weight_scale, input_scale) would be adjusted in different places in the code.
Other than that it's a single function call from utils that's specific to HIP.